cmd/ib-sriov-cni/main.go

```diff
@@ -183,52 +246,25 @@ func cmdAdd(args *skel.CmdArgs) error {
 		}
 	}()
 	if err != nil {
-		return fmt.Errorf("failed to set up pod interface %q from the device %q: %v", args.IfName, netConf.Master, err)
+		return err
 	}
 
-	// run the IPAM plugin
-	if netConf.IPAM.Type != "" {
-		if netConf.IPAM.Type == ipamDHCP {
-			return fmt.Errorf("ipam type dhcp is not supported")
-		}
-		var r types.Result
-		r, err = ipam.ExecAdd(netConf.IPAM.Type, args.StdinData)
-		if err != nil {
-			return fmt.Errorf("failed to set up IPAM plugin type %q from the device %q: %v",
-				netConf.IPAM.Type, netConf.Master, err)
-		}
+	result := &current.Result{}
+	result.Interfaces = []*current.Interface{{
+		Name:    args.IfName,
+		Sandbox: netns.Path(),
+	}}
 
+	if netConf.IPAM.Type != "" {
+		err = runIPAMplugin(args, netConf, netns, &result)
 		defer func() {
 			if err != nil {
 				_ = ipam.ExecDel(netConf.IPAM.Type, args.StdinData)
```

Comment:
What will happen now if runIPAMplugin fails early ? there is no cleanup needed in this case right ?

Reply:
this is similar to my comment above about `doVfConfiguration`  if its assured that nothing bad will happen, im fine with keeping it as is